magnum-kyo/src/test/scala/com/augustnagro/magnum/magkyo/ImmutableRepoKyoTests.scala

```diff
@@ -0,0 +1,197 @@
+package com.augustnagro.magnum.magkyo
+
+import com.augustnagro.magnum.*
+import munit.{FunSuite, Location}
+import kyo.{Id as _, *}
+
+import java.sql.Connection
+import java.time.OffsetDateTime
+import scala.util.{Success, Using}
+
+def immutableRepoKyoTests(
+    suite: FunSuite,
+    dbType: DbType,
+    xa: () => TransactorKyo
+)(using
+    Location,
+    DbCodec[OffsetDateTime]
+): Unit =
+  import suite.*
+
+  def runEffect[A](io: A < (Abort[Throwable] & Async)): A =
```

Comment:
you can use `runLiftTest` https://github.com/getkyo/kyo/blob/2f561cda15104947209498e0fed69d4926c40931/kyo-direct/shared/src/main/scala/kyo/TestSupport.scala#L7-L12
to write better tests (based on kyo-direct)